### PR TITLE
Feat: add variations of delete action for invite service

### LINKED ISSUE #57: Fix: dont pop object in delete method in mock repository
Deleted_at field is always None in mock_repositories now. Fix ASAP.

### LINKED ISSUE #62: Fix: handle optional fields correctly in proto
Protobuf sets a default value for null such as:
* "" for string
* 0 for int
* 1970-01-01 for datetime
Handle it

### PR DESCRIPTION
**Pull request by Nhsdkk**
1) Add delete invite by event id
2) Add delete invite by invitee id
3) Add delete invite by author id
4) Rename delete_invite to delete_invite_by_id
5) Add InviteStatus enum
6) Fix converting errors
7) Add optional invite status to queries to filter invites

### CODE REVIEW

File: invite_service/repository/mock_invite_repository.py [hunk 1]
```diff
@@ -161,6 +183,7 @@ async def create_invite(self, invite: Invite) -> None:
 
         """
         invite.id = str(uuid4())
+        invite.created_at = datetime.now()
```
Comment (ZixMai) [inline invite_service/repository/mock_invite_repository.py]: add "invite.deleted_at = None" at the next line

**Review by ZixMai — CHANGES_REQUESTED** [review]

File: invite_service/repository/mock_invite_repository.py (hunk 1 above)
Comment (Nhsdkk) [reply invite_service/repository/mock_invite_repository.py]: fixed

**Comment by Nhsdkk** [comment]
#62 was fixed in [84e6571](https://github.com/UnspokenTeam/Calendar/pull/61/commits/84e657198ac2e8e4eb3b006ce9a64cd80231169b)

**Review by MaxPlays35 — APPROVED** [review]
Good

**Review by ZixMai — APPROVED** [review]

**Pull request merged**